### PR TITLE
[Feature] Encapsulate interface logic

### PR DESCRIPTION
**Pull request by Tastyep**
# Description

Encapsulate popup creation.
Provide a function for aligning text relatively to its container.
Provide a function for pretty formatting tables.
Use `strings.format` instead of the `..` operator

Make it easier for implementing new popups. 

I add to change the banner because it used non-ASCII characters which caused invalid text alignment.
I used [this website](http://www.patorjk.com/software/taag/#p=testall&f=Graffiti&t=LunarVim), pick another one if you prefer.

## How Has This Been Tested?

`<space>Li`

Small:
image

Medium:
image

Big:
image

No LSP
image

### CODE REVIEW

File: lua/interface/text.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+local M = {}
+
+--- Center align lines relatively to its container
```
Comment (Tastyep) [inline lua/interface/text.lua:3]: to its container  -> to the parent container

**Review by kylo252 — APPROVED** [review]
LGTM!

File: lua/core/info.lua [hunk 2]
```diff
@@ -136,67 +57,89 @@ function M.toggle_popup(ft)
...
+    fmt("* Associated server:    %s", client_name),
+    fmt("* Active:               %s (id: %d)", tostring(is_client_active), client_id),
+    fmt("* Supports formatting:  %s", tostring(document_formatting)),
   }
-  vim.list_extend(buf_lines, lsp_info)
+  local caps_text = "* Capabilities list:    "
+  local caps_text_len = caps_text:len()
+  local enabled_caps = text.format_table(client_enabled_caps, 3, " | ")
+  enabled_caps = text.shift_left(enabled_caps, caps_text_len)
+  enabled_caps[1] = fmt("%s%s", caps_text, enabled_caps[1]:sub(caps_text_len + 1))
```
Comment (abzcoding) [inline lua/core/info.lua:84]: if you mistakenly open it in terminal 😅 
Screen Shot 2021-08-16 at 3 39 35 PM

Comment (abzcoding) [reply lua/core/info.lua:84]: also if i open a `test.fish` file and run the same command 😢

File: lua/interface/popup.lua [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  local default_layout = {
+    relative = "editor",
+    height = math.ceil(vim.o.lines * 0.9),
+    width = math.floor(vim.o.columns * 0.8),
+    style = "minimal",
+    border = "rounded",
+  }
+  default_layout.col = (vim.o.columns - default_layout.width) / 2
+  default_layout.row = (vim.o.lines - default_layout.height) / 2
```
Comment (PasiBergman) [inline lua/interface/popup.lua:24]: I.e. if height calculation is change, then L24 should be something like
```lua
default_layout.row = (vim.o.lines - default_layout.height) / 2 - vim.o.cmdheight
```

File: lua/core/info.lua [hunk 4]
```diff
@@ -1,128 +1,49 @@
...
+local fmt = string.format
+
 local function str_list(list)
-  return "[ " .. table.concat(list, ", ") .. " ]"
+  return fmt("[ %s ]", table.concat(list, ", "))
 end
 
 local function get_formatter_suggestion_msg(ft)
   local null_formatters = require "lsp.null-ls.formatters"
   local supported_formatters = null_formatters.list_available(ft)
```
Comment (PasiBergman) [inline lua/core/info.lua:20]: Maybe we could add condition after L20
```lua
if #supported_formatters == 0 then
  return { }
end
```

File: lua/core/info.lua [hunk 5]
```diff
@@ -1,128 +1,49 @@
...
+    "* Configured formatter needs to be installed and executable.",
+    fmt("* Enable installed formatter(s) with following config in %s", USER_CONFIG_PATH),
     "",
+    fmt("  lvim.lang.%s.formatters = { { exe = '%s' } }", ft, table.concat(supported_formatters, "│")),
   }
 end
 
 local function get_linter_suggestion_msg(ft)
   local null_linters = require "lsp.null-ls.linters"
   local supported_linters = null_linters.list_available(ft)
```
Comment (PasiBergman) [inline lua/core/info.lua:34]: Maybe we could add condition after L34
```lua
if #supported_linters == 0 then
  return { }
end
```

File: lua/core/info.lua (hunk 2 above)
Comment (PasiBergman) [inline lua/core/info.lua:84]: `enabled_caps` could be empty => cannot index [1]

**Review by PasiBergman — CHANGES_REQUESTED** [review]

File: lua/core/info.lua (hunk 2 above)
Comment (abzcoding) [reply lua/core/info.lua:84]: https://github.com/LunarVim/LunarVim/pull/1320#discussion_r689443364

**Review by PasiBergman — APPROVED** [review]

**Pull request merged**